pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -202,9 +202,10 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 
 	// Check whether swap is enabled. The Kubelet does not support running with swap enabled.
 	swapData, err := ioutil.ReadFile("/proc/swaps")
-	if err != nil {
+	if failSwapOn && err != nil {
```

Comment:
What if the error is a permission issue? If we set `failSwapOn` false, the right error can not be returned correctly.

Actually we've already got check in L214 to see whether swap on is supported.

I don't think this change is needed.

Reply:
if failSwapOn is false, the err we need ignore but not return err always.